components/terms_of_service/terms_of_service.tsx

```diff
@@ -19,19 +19,40 @@ import {browserHistory} from 'utils/browser_history';
 import messageHtmlToComponent from 'utils/message_html_to_component';
 import {formatText} from 'utils/text_formatting';
 import {Constants} from 'utils/constants.jsx';
+import EmojiMap from 'utils/emoji_map';
 
-export default class TermsOfService extends React.PureComponent {
-    static propTypes = {
-        location: PropTypes.object,
-        termsEnabled: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            getTermsOfService: PropTypes.func.isRequired,
-            updateMyTermsOfServiceStatus: PropTypes.func.isRequired,
-        }).isRequired,
-        emojiMap: PropTypes.object.isRequired,
+interface UpdateMyTermsOfServiceStatusResponse {
+    terms_of_service_create_at: number;
+    terms_of_service_id: string;
+    user_id: number;
+}
+
+export interface TermsOfServiceProps {
+    location: {search: string},
+    termsEnabled: boolean,
+    actions: {
+        getTermsOfService: () => Promise<{ data: ReduxTermsOfService }>;
+        updateMyTermsOfServiceStatus: (
+            termsOfServiceId: string,
+            accepted: boolean
+        ) => {data: UpdateMyTermsOfServiceStatusResponse};
     };
+    emojiMap: EmojiMap;
+}
+
+interface TermsOfServiceState {
+    customTermsOfServiceId: string;
+    customTermsOfServiceText: string;
+    loading: boolean;
+    loadingAgree: boolean;
+    loadingDisagree: boolean;
+    serverError: JSX.Element | null;
```

Comment:
This can be made a bit simpler by using `React.ReactNode` instead.
```suggestion
    serverError: React.ReactNode;
```

Reply:
Done